pkg/storage/batch.go

```diff
@@ -751,6 +774,77 @@ func fetchLazyChunks(ctx context.Context, s chunk.SchemaConfig, chunks []*LazyCh
 	return nil
 }
 
+func postFetchFilter(ctx context.Context, chunks []chunk.Chunk, filterer PostFetcherChunkFilterer, schema chunk.SchemaConfig) ([]chunk.Chunk, error) {
+	if len(chunks) == 0 {
+		return chunks, nil
+	}
+	postFilterChunkLen := 0
+	log, ctx := spanlogger.New(ctx, "Batch.ParallelPostFetchFilter")
+	log.Span.LogFields(otlog.Int("chunks", len(chunks)))
+	defer func() {
+		log.Span.LogFields(otlog.Int("postFilterChunkLen", postFilterChunkLen))
+		log.Span.Finish()
+	}()
+
+	logSelector, err := logql.ParseLogSelector(filterer.Request().Selector, true)
+	if err != nil {
+		return nil, err
+	}
+	if !logSelector.HasFilter() {
+		return chunks, nil
+	}
+	pipeline, err := logSelector.Pipeline()
+	if err != nil {
+		return nil, err
+	}
+	//
+	streamPipeline := pipeline.ForStream(chunks[0].Metric.WithoutLabels(labels.MetricName))
+	result := make([]chunk.Chunk, 0)
+
+	for _, cnk := range chunks {
+		chunkData := cnk.Data
+		lokiChunk := chunkData.(*chunkenc.Facade).LokiChunk()
+		blocks := lokiChunk.Blocks(filterer.Request().Start, filterer.Request().End)
+		if len(blocks) == 0 {
+			continue
+		}
+		postFilterChunkData := chunkenc.NewMemChunk(filterer.Encoding(), filterer.HeadBlockFmt(), cnk.Data.Size(), cnk.Data.Size())
+		for _, block := range blocks {
+			iterator := block.Iterator(ctx, streamPipeline)
+			//1: post filter
+			for iterator.Next() {
+				entry := iterator.Entry()
+				err := postFilterChunkData.Append(&entry)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+		if postFilterChunkData.Size() == 0 {
+			continue
+		}
+		if err := postFilterChunkData.Close(); err != nil {
+			return nil, err
+		}
+		firstTime, lastTime := loki_util.RoundToMilliseconds(postFilterChunkData.Bounds())
+
+		postFilterCh := chunk.NewChunk(
+			cnk.UserID, cnk.Fingerprint, cnk.Metric,
+			chunkenc.NewFacade(postFilterChunkData, filterer.BlockSize(), filterer.TargetChunkSize()),
```

Comment:
You don't need to set blocksize and target chunk size since you are not writing any data this point onwards.

Reply:
👍，It works fine even if it is set to 0